File: RepoTransBench/target_projects/Java/Go/spotify_docker-maven-plugin/public_tests/abstract_docker_mojo_public_test.go
package public_tests

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

type AbstractDockerMojo struct{}
func (a *AbstractDockerMojo) ReplaceRegistryUrl(orig, to string) string {
	return to
}

func TestRegistryUrlReplacePublic(t *testing.T) {
	mojo := &AbstractDockerMojo{}
	url := mojo.ReplaceRegistryUrl("index.docker.io", "my.other.io")
	assert.Equal(t, "my.other.io", url)
}

File: RepoTransBench/source_projects/Java/spotify_docker-maven-plugin/src/test/java/com/spotify/docker/AbstractDockerMojoPublicTest.java
package com.spotify.docker;

import org.junit.Test;
import static org.junit.Assert.*;

public class AbstractDockerMojoPublicTest {

    @Test
    public void testRegistryUrlReplacePublic() {
        AbstractDockerMojo mojo = new AbstractDockerMojo() {};
        String url = mojo.replaceRegistryUrl("index.docker.io", "my.other.io");
        assertEquals("my.other.io", url);
    }

    // The method is protected; additional more complex logic would need subclassing as needed
}